flink-core/src/main/java/org/apache/flink/configuration/NettyShuffleEnvironmentOptions.java

```diff
@@ -379,6 +380,28 @@ public class NettyShuffleEnvironmentOptions {
                                     + "tasks have occupied all the buffers and the downstream tasks are waiting for the exclusive buffers. The timeout breaks"
                                     + "the tie by failing the request of exclusive buffers and ask users to increase the number of total buffers.");
 
+    @Documentation.Section(Documentation.Sections.ALL_TASK_MANAGER_NETWORK)
+    @Experimental
+    @Internal
+    public static final ConfigOption<Boolean> NETWORK_HYBRID_SHUFFLE_ENABLE_TIERED_STORAGE =
+            ConfigOptions.key("taskmanager.network.hybrid-shuffle.enable-tiered-storage")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "The option is used to enable tiered storage architecture for hybrid shuffle mode.");
```

Comment:
1. This is not internal.
2. JavaDoc is missing.
3. What is "tiered storage architecture" from the user's perspective? How is it different from the original architecture? What does the user needs to know about this?
4. Why is the default value `false`? How can a user decide which mode to use? Would there be anyone enable it at all?